File: vane-proxy-core/src/main/java/org/oddlama/vane/proxycore/config/AuthMultiplex.java
package org.oddlama.vane.proxycore.config;

import java.util.*;
import java.util.stream.Collectors;

public class AuthMultiplex {

    public Integer port;
    private final List<UUID> allowedUuids;

    public AuthMultiplex(Integer port, List<String> allowedUuids) {
        this.port = port;

        if (allowedUuids == null || allowedUuids.isEmpty()) {
            this.allowedUuids = List.of();
        } else {
            this.allowedUuids = allowedUuids
                .stream()
                .filter(s -> !s.isEmpty())
                .map(UUID::fromString)
                .collect(Collectors.toList());
        }
    }

    public boolean uuidIsAllowed(UUID uuid) {
        return allowedUuids.isEmpty() || allowedUuids.contains(uuid);
    }
}

File: vane-proxy-core/src/main/kotlin/org/oddlama/vane/proxycore/config/AuthMultiplex.kt
package org.oddlama.vane.proxycore.config

import java.util.*
import java.util.stream.Collectors

class AuthMultiplex(@JvmField var port: Int?, allowedUuids: MutableList<String?>?) {
    private val allowedUuids: MutableList<UUID?>

    init {
        if (allowedUuids.isNullOrEmpty()) {
            this.allowedUuids = mutableListOf()
        } else {
            this.allowedUuids = allowedUuids
                .stream()
                .filter { s: String? -> !s!!.isEmpty() }
                .map { name: String? -> UUID.fromString(name) }
                .collect(Collectors.toList())
        }
    }

    fun uuidIsAllowed(uuid: UUID?): Boolean {
        return allowedUuids.isEmpty() || allowedUuids.contains(uuid)
    }
}
